### DISCUSSION TITLE
vCenter Custom Attribute

### DISCUSSION THREAD

**Discussion by maw1146** [bb-Ricardo/netbox-sync#113] [Ideas]
Hi,
we would like to sync custom attribute from vCenter to a custom attribute in Netbox.
Do you know if it's possible and/or if it's planned to be developed?

Our idea is to have kind of a mapping. So create a custom attribute in vCenter called "attribute1" and sync with the custom attribute of Netbox called "attribute1".
(in our case, both attributes contain text string)

Thanks for the great tool and support :)

**Comment by bb-Ricardo** [comment]
Hi, without looking at it I would say it is possible. Can add a screenshot from a device in vCenter which has a custom attribute?

**Comment by maw1146** [comment]
Hi,
sure, attached the screeshot.

Just realized I was not clear in my first comment. I meant a custom attribute of the VM.

In the screenshot you can see the custom attribute "BackupLog" with the text written by the backup system.
We would like to sync that text to Netbox :)

![vCenterCustomAttribute](https://user-images.githubusercontent.com/17356508/139457998-673b0f50-867b-4ab3-8d62-49e9d225a289.jpg)

**Reply by bb-Ricardo** [reply]
Actually seems easy enough with the current tooling. Will see what I can do the next few days.

**Reply by maw1146** [reply]
that seems perfect, thanks a lot :)

**Reply by bb-Ricardo** [reply]
Hi, I added https://github.com/bb-Ricardo/netbox-sync/tree/feature/read-custom-attributes

you can checkout this branch, add this to your config:
https://github.com/bb-Ricardo/netbox-sync/blob/ce5eb04ff82d1b036dd8898b6e28bde4082dae9f/settings-example.ini#L235-L238

and sync again.

Let me know if it works.

**Reply by maw1146** [reply]
Hey! thanks a lot! it's working perfectly! :)

**Reply by bb-Ricardo** [reply]
Thank you for testing
